src/thermo/include/antioch/nasa7_curve_fit.h

```diff
@@ -175,14 +175,19 @@ namespace Antioch
 
     for (unsigned int i=begin_interval; i != end_interval; ++i)
       {
-         const CoeffType *a = this->coefficients(interval);
+         const CoeffType *a = this->coefficients(i);
+
+         typedef typename Antioch::raw_value_type<StateType>::type raw_type;
 
          /* h/RT = a0     + a1*T/2 + a2*T^2/3 + a3*T^3/4 + a4*T^4/5 + a5/T */
         returnval = Antioch::if_else
         ( interval == i,
            StateType(  a[0] +
-                       a[1]*cache.T/2.0L + a[2]*cache.T2/3.0L + a[3]*cache.T3/4.0L +
-                       a[4]*cache.T4/5.0L + a[5]/cache.T),
+                       a[1]*cache.T/raw_type(2) +
```

Comment:
I got a compiler error with VexCL (all the others were fine). I'll reproduce the error and post here.